src/Hosting/ResponsivePageRouteModelConvention.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+using Wangkanai.Detection.Models;
+
+namespace Wangkanai.Detection.Hosting
+{
+    internal class ResponsivePageRouteModelConvention : IPageRouteModelConvention
+    {
+        public void Apply(PageRouteModel model)
+        {
+            var fileNameWithoutExtension = model.ViewEnginePath.Split('/')[^1];
+            if (!fileNameWithoutExtension.Contains('.'))
+            {
+                // Normal page, nothing to do!
+                return;
+            }
+
+            // This just implements the 'suffix' strategy
+            var split = fileNameWithoutExtension.Split('.');
+            if (split.Length != 2)
```

Comment:
@rynowak I have a few questions.

1. Would effect other views in the asp.net core ecosystem and do you have a specification for it? 
2. Does the suffix mean only for the file extension like `index.mobile.cshtml`?
3. Can the split array have more than 2? Maybe like `index.ie.mobile.cshhtml`?

Sorry if I ask too much. Thank you for your much.

Reply:
> Would effect other views in the asp.net core ecosystem and do you have a specification for it?

No, there's no convention for this. I just tried to make it do what you were doing in the sample. Feel free to change this to meet your needs.

> Does the suffix mean only for the file extension like index.mobile.cshtml

Sorry, I'm not sure what this means. If you have questions about what these values represent, I'd suggest setting a breakpoint and running your sample.

> Can the split array have more than 2? Maybe like index.ie.mobile.cshhtml?

Sure, this could be whatever convention you want. I coded this up based on what I saw the sample doing.

Reply:
Another questions I see ASP.NET Core using `ApplicationModels`, this is so that conflict with the people using the framework of the convention of `Models` and `ViewModels` folder right? Is this by my understanding if that is correct.

Reply:
> Another questions I see ASP.NET Core using ApplicationModels, this is so that conflict with the people using the framework of the convention of Models and ViewModels folder right? Is this by my understanding if that is correct.

These things are totally unrelated, I think the term convention has a few uses which is causing confusion.

The types in `.ApplicationModels` allow you to customize the data about your application gather by MVC. So you can customize the framework without replacing framework features, you just change the data (in our example, you change what the route is).

The convention of using `Models` and `ViewModels` as folders is just a naming style that people like. There's **NO** framework behavior related to that.

Reply:
Thank you very much. Your explanation make things a lot clearer to me now.